### DISCUSSION TITLE
machine.SDCard not on UM_FEATHERS2 ?

### DISCUSSION THREAD

**Discussion by SquirtleSquadLeader** [micropython/micropython#14266] [ESP32] [answered]
Can someone confirm if machine.SDCard() is included in the UM_FEATHERS2 port?  This board does not have a microSD card reader.  I was hoping to use an external microSD card breakout.  I am unable to import it using the downloaded v1.22.2.bin . 

Trying to perform an import yields the following output:

```
MPY: soft reboot                                                                
MicroPython v1.22.2 on 2024-02-22; FeatherS2 with ESP32-S2                      
Type "help()" for more information.                                             
>>> from machine import SDCard                                                  
Traceback (most recent call last):                                              
  File "<stdin>", line 1, in <module>                                           
ImportError: can't import name SDCard                                           
>>> 
```

**Comment by davefes** [comment]
At `>>> help('modules')`

**Comment by sosi-deadeye** [comment, marked as answer]
I don't know why, but the hardware `SDCard` support is disabled: https://github.com/micropython/micropython/blob/master/ports/esp32/boards/UM_FEATHERS2/mpconfigboard.h#L6

You could try to compile it with support or just use the Python version of [SDCard](https://github.com/micropython/micropython-lib/tree/master/micropython/drivers/storage/sdcard).

If the ESP32S2 is connected to the internet, you could use `mip` in the `repl`.

```python
import mip


mip.install("sdcard")
```

The __init__ method: https://github.com/micropython/micropython-lib/blob/master/micropython/drivers/storage/sdcard/sdcard.py#L42

**Reply by SquirtleSquadLeader** [reply]
Thank you.  I was able to get micropython-lib/micropython/drivers/storage/sdcard to work.  

I am interested in recompiling with SDCard support enabled.  Can you advise on what edits would need to be made to enable SDCard to be part of the .bin? 

I am using an Adafruit Adalogger featherwing which would put my chip select on GPIO 3.  I presume the below would work?


```
#define MICROPY_HW_BOARD_NAME "FeatherS2"
#define MICROPY_HW_MCU_NAME "ESP32-S2"
#define MICROPY_PY_NETWORK_HOSTNAME_DEFAULT "FeatherS2"

#define MICROPY_PY_BLUETOOTH                (0)
#define MICROPY_HW_ENABLE_SDCARD            (3)

#define MICROPY_HW_I2C0_SCL (9)
#define MICROPY_HW_I2C0_SDA (8)

#define MICROPY_HW_SPI1_MOSI (35)  // SDO
#define MICROPY_HW_SPI1_MISO (37)  // SDI
#define MICROPY_HW_SPI1_SCK (36)

```
